src/sources/http_server.rs

```diff
@@ -122,11 +122,60 @@ pub struct SimpleHttpConfig {
     #[configurable(derived)]
     #[serde(default, deserialize_with = "bool_or_struct")]
     acknowledgements: SourceAcknowledgementsConfig,
+
+    /// The namespace to use for logs. This overrides the global setting.
+    #[configurable(metadata(docs::hidden))]
+    #[serde(default)]
+    pub log_namespace: Option<bool>,
 }
 
-impl GenerateConfig for SimpleHttpConfig {
-    fn generate_config() -> toml::Value {
-        toml::Value::try_from(Self {
+impl SimpleHttpConfig {
+    /// Builds the `schema::Definition` for this source using the provided `LogNamespace`.
+    fn schema_definition(&self, log_namespace: LogNamespace) -> Definition {
+        let mut schema_definition = self
+            .decoding
+            .as_ref()
+            .unwrap_or(&default_decoding())
+            .schema_definition(log_namespace)
+            .with_standard_vector_source_metadata()
+            .with_source_metadata(
+                SimpleHttpConfig::NAME,
+                parse_value_path(&self.path_key)
+                    .ok()
+                    .map(LegacyKey::Overwrite),
+                &owned_value_path!("path"),
+                Kind::bytes(),
+                None,
+            )
+            // for metadata that is added to the events dynamically from the self.headers
```

Comment:
Thinking about this again, it's _somewhat_ of an anti-pattern (at least from my POV, so far) that this enrichment is configurable. Probably should gather input from others, but it may make sense to add all headers to the metadata when using the vector namespace.

Reply:
@neuronull did you have any thoughts here, or did you want to open a follow-up issue to track this line of though?

Reply:
My original take on this if I recall, was to loop over the user defined headers and add each to the vector namespace.

Don't feel like I have enough of the RFC/goal for this effort in my veins to be able to argue one way or the other.

This was one of the items I was going to bring-up today with Nathan but I forgot he is out :(

Reply:
👍 I think leaving it as is, and marking it as a thing to discuss down the line.

Reply:
I'll merge as-is, and make a note to discuss this next week. If I need to open a small PR to adjust it'll be NBD.

Reply:
Just noting here that we discussed it with Nathan and he agrees with the selected approach in this case.

> This one seems reasonable to me, mainly because an HTTP request can have quite a few headers and many of them are usually ignored. If theres only a couple and they are usually useful, I'd probably default to always including them (with the vector namespace)